lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQuerySqlProvider.java

```diff
@@ -72,7 +72,9 @@ public static String queryTaskSummaries() {
         + "<if test='selectAndClaim == true'> "
         + "FETCH FIRST ROW ONLY FOR UPDATE"
         + "</if>"
-        + "<if test=\"_databaseId == 'db2'\">WITH RS USE AND KEEP UPDATE LOCKS </if>"
+        + "<if test=\"_databaseId == 'db2' and selectAndClaim \">WITH RS USE "
```

Comment:
like this the select and claim wouldn't work for non db2 databases anymore. Using the old syntax, but nesting the second if would fix the issue with compatibility for other databases, what do you think?